src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -457,6 +458,9 @@ def python_type_to_snow_type(tp: Union[str, Type]) -> Tuple[DataType, bool]:
                 False,
             )
 
+    if tp == SnowparkDataFrame:
+        return StructType([]), False
```

Comment:
Curious: why is this an empty list? Is this for the case of dynamic table return type?

Reply:
+1. Can we input nothing if it's empty?

Reply:
`StructType` requires one positional argument for creating a StructType object. This is indeed the case of dynamic return type.